src/lib/form-field/form-field.md

```diff
@@ -0,0 +1,129 @@
+by adding a `<mat-placeholder>` element inside the form field. Only one of these options should be
+used, specifying both will raise an error.
+
+If the form field control is marked with a `required` attribute, an asterisk will be appended to the
+placeholder to indicate the fact that it is a required field. If unwanted, this can be disabled by
+setting the `hideRequiredMarker` property on `<mat-form-field>`
+
+The `floatPlaceholder` property of `<mat-form-field>` can be used to change this default floating
+behavior. It can set to `never` to hide the placeholder instead of float it when text is present in
+the form field control. It can be set to `always` to float the placeholder even when no text is
+present in the form field control. It can also be set to `auto` to restore the default behavior.
+
+<!-- example(form-field-placeholder) -->
+
+Global default placeholder options can be specified by setting providing a value for
+`MAT_PLACEHOLDER_GLOBAL_OPTIONS` in your application's root module. Like the property, the global
+setting can be either `always`, `never`, or `auto`.
+
+```ts
+@NgModule({
+  providers: [
+    {provide: MAT_PLACEHOLDER_GLOBAL_OPTIONS, useValue: {float: 'always'}}
+  ]
+})
+```
+
+### Hint labels
+Hint labels are additional descriptive text that appears below the form field's underline. A
+`<mat-form-field>` can have up to two hint labels; one start-aligned (left in an LTR language, right
+in RTL), and one end-aligned.
+
+Hint labels are specified in one of two ways: either by using the `hintLabel` property of
+`<mat-form-field>`, or by adding a `<mat-hint>` element inside the form field. When adding a hint
+via the `hintLabel` property, it will be treated as the start hint. Hints added via the
+`<mat-hint>` hint element can be added to either side by setting the `align` property on
+`<mat-hint>` to either `start` or `end`. Attempting to add multiple hints to the same side will
+raise an error.
+
+<!-- example(form-field-hint) -->
+
+### Error messages
+Error messages can be shown under the form field underline by adding `mat-error` elements inside the
+form field. Errors are hidden initially and will be displayed on invalid form fields after the user
+has interacted with the element or the parent form has been submitted. Since the errors occupy the
+same space as the hints, the hints are hidden when the errors are shown.
+
+If an form field can have more than one error state, it is up to the consumer to toggle which
+messages should be displayed. This can be done with CSS, `ngIf` or `ngSwitch`. Multiple error
+messages can be shown at the same time if desired, but the `<mat-form-field>` only reserves enough
+space to display one error message at a time. Ensuring that enough space is available to display
+multiple errors is up to the user.
+
+<!-- example(form-field-error) -->
+
+### Prefix & suffix
+HTML can be included before and after the input tag, as a prefix or suffix. It will be underlined
```

Comment:
`It will be underlined...` -> `It will be included within the visual container that wraps the form control`.

Since the new style inputs are more than just the underline style